### PR TITLE
fix: Resolve bug with declaring Public IP Prefix use on managementIP

### LINKED ISSUE #1867: [AVM Module Issue]: Azure-Firewall PIP issue during deployment
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/network/azure-firewall

### (Optional) Module Version

0.2.0

### Description

With the following bicep config

```bicep
module deployazurefw 'br/public:avm/res/network/azure-firewall:0.2.0' = {
  scope: resourceGroup('rg-${ResourcePrefix}-platform-01-connectivity-01')
  name: 'deployafw'
  params: {
    name: 'afw-${ResourcePrefix}-platform-01'
    location: deploymentLocation
    azureSkuTier: 'Basic'
    virtualNetworkResourceId: hubvnetresource
    publicIPAddressObject: {
      name: 'pip-platform-${ResourcePrefix}-afw-01'
      publicIPAllocationMethod: 'Static'
      publicIPPrefixResourceId: publicipprefix
      skuName: 'Standard'
      skuTier: 'Regional'
      zones: []
    }
    managementIPAddressObject: {
      name: 'pip-platform-${ResourcePrefix}-afw-mgmt-01'
      publicIPAllocationMethod: 'Static'
      publicIPPrefixResourceId: publicipprefix
      skuName: 'Standard'
      skuTier: 'Regional'
      zones: []
    }
    firewallPolicyId: deployafwp.outputs.resourceId
  }
}
```
When re-running a deployment which includes the above the following error is generated.

``
Microsoft.Network/publicIPAddresses/pip-platform-twr-afw-01 has an existing availability zone constraint NoZone and the request has availability zone constraint 1, 2, 3, which do not match. Zones cannot be added/updated/removed once the resource is created. The resource cannot be updated from regional to zonal or vice-versa. (Code: ResourceAvailabilityZonesCannotBeModified)
``
Despite there being no changes made to the configuration when re-deploying. I'm assuming there is something in the template attempting to change the Public IP config?
Also, the template ignores using the specified public IP prefix for the management IP address and creates it's own separate public IP (with correct naming as above).

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by hundredacres**
## Description

Resolved issue when specifying a public IP prefix for the management IP address. Updated to use latest PublicIPAddress AVM module. Updated API version of Microsoft.Network/publicIPAddresses used in tests. Updated formatting of zone param default values. Also added new e2e tests for Public IP Prefix usage.

Fixes #1867 
Closes #1867 

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.network.azure-firewall](https://github.com/hundredacres/bicep-registry-modules/actions/workflows/avm.res.network.azure-firewall.yml)       |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [X] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [X] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/network/azure-firewall/version.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
-    "version": "0.2",
+    "version": "0.2.1",
```
Comment (AlexanderSehr) [inline avm/res/network/azure-firewall/version.json:3]: Hey @hundredacres, please note that patch/bug level updates are automatically applied by the CI. In other wrods, please undo, or update the minor ;)

File: avm/res/network/azure-firewall/tests/e2e/publicipprefix/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    location: resourceLocation
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/network/azure-firewall/tests/e2e/publicipprefix/main.test.bicep:40]: Used?

File: avm/res/network/azure-firewall/version.json (hunk 1 above)
Comment (hundredacres) [reply avm/res/network/azure-firewall/version.json]: @AlexanderSehr  I revved the minor.

File: avm/res/network/azure-firewall/tests/e2e/publicipprefix/main.test.bicep (hunk 2 above)
Comment (hundredacres) [reply avm/res/network/azure-firewall/tests/e2e/publicipprefix/main.test.bicep]: @AlexanderSehr  Removed from the test since it's not used.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**